esp/xslt/esdl2diffecl.xslt

```diff
@@ -0,0 +1,813 @@
+  </xsl:if>
+  <xsl:if test="@ecl_type and (@type or @complex_type)">
+        <xsl:choose>
+         <xsl:when test="name()='EsdlArray'">
+           <xsl:text> // Real type: </xsl:text>
+           <xsl:text>dataset(</xsl:text>
+           <xsl:value-of select="@type"/>
+           <xsl:text>) </xsl:text>
+           <xsl:call-template name="output_ecl_name"/>
+           <xsl:text> {xpath('</xsl:text>
+           <xsl:value-of select="@name"/>
+           <xsl:text>/</xsl:text>
+           <xsl:call-template name="output_item_tag"/>
+           <xsl:text>')};</xsl:text>
+         </xsl:when>
+         <xsl:when test="name()='EsdlElement' and starts-with(@ecl_type,'tns:')">
+           <xsl:text> // Real type: RECORD </xsl:text>
+           <xsl:value-of select="@type|@complex_type"/>
+         </xsl:when>
+         <xsl:when test="name()='EsdlElement' and not(starts-with(@ecl_type,'tns:'))">
+           <xsl:text> // Xsd type: </xsl:text>
+           <xsl:value-of select="@type|@complex_type"/>
+         </xsl:when>
+         <xsl:otherwise>
+             <xsl:value-of select="@type"/>
+         </xsl:otherwise>
+        </xsl:choose>
+    </xsl:if>
+</xsl:template>
+
+<xsl:template name="output_xpath">
+  <xsl:text>xpath('</xsl:text>
+  <xsl:choose>
+    <xsl:when test="@ecl_path"><xsl:value-of select="@ecl_path"/></xsl:when>
+    <xsl:when test="@get_data_from"><xsl:if test="@attribute"><xsl:value-of select="'@'"/></xsl:if> <xsl:value-of select="@get_data_from"/></xsl:when>
+    <xsl:when test="@alt_data_from"><xsl:if test="@attribute"><xsl:value-of select="'@'"/></xsl:if> <xsl:value-of select="@alt_data_from"/></xsl:when>
+    <xsl:otherwise><xsl:if test="@attribute"><xsl:value-of select="'@'"/></xsl:if> <xsl:value-of select="@name"/></xsl:otherwise>
+  </xsl:choose>
+  <xsl:text>')</xsl:text>
+</xsl:template>
+
+<xsl:template name="output_item_tag">
+    <xsl:choose>
+         <xsl:when test="@item_tag"><xsl:value-of select="@item_tag"/></xsl:when>
+         <xsl:when test="@type and (@type='int' or @type='integer' or @type='bool' or @type='short' or @type='float' or @type='double' or @type='string' or @type='long' or @type='decimal' or @type='byte' or @type='unsignedInt' or @type='unsignedShort' or @type='unsignedByte')">
+           <xsl:value-of select="'Item'"/>
+         </xsl:when>
+         <xsl:when test="@type">
+           <xsl:value-of select="@type"/>
+         </xsl:when>
+         <xsl:otherwise><xsl:value-of select="'Item'"/></xsl:otherwise>
+    </xsl:choose>
+</xsl:template>
+
+<xsl:template name="output_active_check">
+  <xsl:choose>
+    <xsl:when test="_diff_selectors/selector">
+      <xsl:for-each select="_diff_selectors/selector">
+        <xsl:if test="position()!=1"><xsl:text> OR</xsl:text></xsl:if><xsl:text> Monitor</xsl:text><xsl:value-of select="."/>
+      </xsl:for-each>
+    </xsl:when>
+    <xsl:otherwise>is_active</xsl:otherwise>
+  </xsl:choose>
+</xsl:template>
+
+</xsl:stylesheet>
```

Comment:
I could only skim over XSLT, it looked reasonable from a high level view, we might need an expert to review. BTW, noticed a few extra lines throughout, but didn't think it was worth pointing out inline.